lib/events/athena/querier.go

```diff
@@ -133,12 +157,33 @@ func (q *querier) SearchSessionEvents(fromUTC, toUTC time.Time, limit int,
 		}
 		filter.condition = condFn
 	}
-	return q.searchEvents(context.TODO(), fromUTC, toUTC, limit, order, startKey, filter, sessionID)
+	events, keyset, err := q.searchEvents(context.TODO(), searchEventsRequest{
+		fromUTC:   fromUTC,
+		toUTC:     toUTC,
+		limit:     limit,
+		order:     order,
+		startKey:  startKey,
+		filter:    filter,
+		sessionID: sessionID,
+	})
+	return events, keyset, trace.Wrap(err)
 }
 
-func (q *querier) searchEvents(ctx context.Context, fromUTC, toUTC time.Time, limit int,
-	order types.EventOrder, startKey string, filter searchEventsFilter, sessionID string,
-) ([]apievents.AuditEvent, string, error) {
+type searchEventsRequest struct {
+	fromUTC, toUTC time.Time
+	limit          int
+	order          types.EventOrder
+	startKey       string
+	filter         searchEventsFilter
+	sessionID      string
+}
+
+func (q *querier) searchEvents(ctx context.Context, req searchEventsRequest) ([]apievents.AuditEvent, string, error) {
+	if q.limiter != nil && !q.limiter.Allow() {
+		// TODO(tobiaszheller): return how much use should sleep at some point.
+		return nil, "", trace.LimitExceeded("rate limit exceeded for searching events")
```

Comment:
nit:
We can consider decoupling SearchEventsLimiter from AtenaAuditLoger: 
```
type SearchEventsLimmiter struct {
  limiter  
  inner events.AuditLogger
}

func  .. EmitAuditEvent(context.Context,  ...) ... { 
   return inner.EmitAuditEvent(...)
}
func  .. SearchSessionEvents(context.Context, ...) ... { 
      if limit {
       return ...
    }
    return inner.SearchSessionEvents(..)
}
func  .. SearchEvents(context.Context, ...) ... { 
      if limit {
       return ...
    }
    return inner.SearchEvents(..)
}
```

Reply:
reworked in https://github.com/gravitational/teleport/pull/24918/commits/dfc304cf14978115ef6f44605b868b9752047c71